plugins/woocommerce/src/Internal/Admin/Agentic/AgenticWebhookManager.php

```diff
@@ -0,0 +1,327 @@
+	 * @return array Modified topics.
+	 */
+	public function register_webhook_topic_names( $topics ) {
+		$topics['action.woocommerce_agentic_order_created'] = __( 'Agentic Order Created', 'woocommerce' );
+		$topics['action.woocommerce_agentic_order_updated'] = __( 'Agentic Order Updated', 'woocommerce' );
+		return $topics;
+	}
+
+	/**
+	 * Handle order creation.
+	 *
+	 * @param int      $order_id Order ID.
+	 * @param WC_Order $order    Order object.
+	 */
+	public function handle_order_created( $order_id, $order ) {
+		if ( ! $this->should_trigger_webhook( $order ) ) {
+			return;
+		}
+
+		/**
+		 * Fires when an Agentic order is created.
+		 *
+		 * @since 10.3.0
+		 *
+		 * @param int      $order_id Order ID.
+		 * @param WC_Order $order    Order object.
+		 */
+		do_action( 'woocommerce_agentic_order_created', $order_id, $order );
+	}
+
+	/**
+	 * Handle order status changes.
+	 *
+	 * @param int      $order_id   Order ID.
+	 * @param string   $old_status Old status.
+	 * @param string   $new_status New status.
+	 * @param WC_Order $order      Order object.
+	 */
+	public function handle_order_status_changed( $order_id, $old_status, $new_status, $order ) {
+		if ( ! $this->should_trigger_webhook( $order ) ) {
+			return;
+		}
+
+		// Prevent duplicate firing for the same status change.
+		$event_key = 'status_' . $order_id . '_' . $old_status . '_' . $new_status;
+		if ( isset( self::$processed_events[ $event_key ] ) ) {
+			return;
+		}
+		self::$processed_events[ $event_key ] = true;
+
+		/**
+		 * Fires when an Agentic order status changes.
+		 *
+		 * @since 10.3.0
+		 *
+		 * @param int      $order_id Order ID.
+		 * @param WC_Order $order    Order object.
+		 */
+		do_action( 'woocommerce_agentic_order_updated', $order_id, $order );
+	}
+
+	/**
+	 * Handle order refunds.
+	 *
+	 * @param int $order_id  Order ID.
+	 * @param int $refund_id Refund ID.
+	 */
+	public function handle_order_refunded( $order_id, $refund_id ) {
+		$order = wc_get_order( $order_id );
+		if ( ! $order || ! $this->should_trigger_webhook( $order ) ) {
+			return;
+		}
+
+		// Prevent duplicate firing for the same refund.
+		$event_key = 'refund_' . $order_id . '_' . $refund_id;
+		if ( isset( self::$processed_events[ $event_key ] ) ) {
+			return;
+		}
+		self::$processed_events[ $event_key ] = true;
+
+		/**
+		 * Fires when an Agentic order is refunded.
+		 *
+		 * @since 10.3.0
+		 *
+		 * @param int      $order_id Order ID.
+		 * @param WC_Order $order    Order object.
+		 */
+		do_action( 'woocommerce_agentic_order_updated', $order_id, $order );
+	}
+
+	/**
+	 * Check if webhook should be triggered for this order.
+	 *
+	 * @param WC_Order $order Order object.
+	 * @return bool True if webhook should be triggered.
+	 */
+	private function should_trigger_webhook( $order ) {
+		// Only trigger for orders with an Agentic checkout session ID.
+		$checkout_session_id = $order->get_meta( '_agentic_checkout_session_id' );
```

Comment:
Nitpick: There will be `OrderMetaKey::AGENTIC_CHECKOUT_SESSION_ID` to use rather than this plain string `_agentic_checkout_session_id`